### PR TITLE
smake: remove ENV.O1 (on Intel only) and deparallelize

### PR DESCRIPTION
**Pull request by carlocab**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

See #70594.

### CODE REVIEW

**Comment by carlocab** [comment]
There's probably no need to get rid of all of the older bottles for this.

**Comment by carlocab** [comment]
License found at the homepage: https://sourceforge.net/projects/s-make/

Looking at the source tarball, only the configure scripts say "at your option", so this is probably just `GPL-2.0-only`.

**Comment by carlocab** [comment]
Maybe the `-O1` flag is still needed on ARM. It built fine for me locally when I dropped the `deparallelize` and optimisation flags...

I'll have a look again at this when the other runners finish.

File: Formula/smake.rb [hunk 1]
```diff
@@ -17,11 +18,8 @@ class Smake < Formula
   end
 
   def install
-    # The bootstrap smake does not like -j
-    ENV.deparallelize
-    # Xcode 9 miscompiles smake if optimization is enabled
-    # https://sourceforge.net/p/schilytools/tickets/2/
-    ENV.O1 if DevelopmentTools.clang_build_version >= 900
+    # ARM fails to build with Os
+    ENV.O1 if Hardware::CPU.arm?
```
Comment (SMillerDev) [inline Formula/smake.rb:22]: This should probably be reported somewhere though I don't know where.

Comment (Bo98) [reply Formula/smake.rb:22]: It was here: https://sourceforge.net/p/schilytools/tickets/2/

It's fixed in smake 1.3 (I think), which is not independently packaged anymore. It's in the latest `schilytools`, though that updates more often than `smake` gets a version bump so you'll need to inspect the source files for the actual version number (and even then there may be stealth changes in newer archives).

**Review by SMillerDev — APPROVED** [review]

**Pull request merged**